internal/controller/stages/stages.go

```diff
@@ -313,6 +313,7 @@ func (r *reconciler) syncStage(
 	}
 
 	status.ObservedGeneration = stage.Generation
+	status.CurrentPromotion = nil
```

Comment:
I know you mentioned it in the PR's description, but I'm not clear about the need for this line.

Reply:
Once I understand, LGTM

Reply:
Nevermind. The operative phrase was "once it has gotten path the no non-terminal promos check."

LGTM

Reply:
It's for resilience. We can't rely solely on the promo reconciler for clearing this because promos can be deleted out of band and never get a chance to clear the promotion.

Originally, I only had it in the stage reconciler, but the stage didnt' get reconciled until several minutes after the promo finished. So I need it there so UI is reactive.

Reply:
> but the stage didnt' get reconciled until several minutes after the promo finished

Ya... slow refresh is a vexing problem when there aren't associated Argo CD Apps. ☹️